### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by chrisza4**

### CODE REVIEW

File: src/poker_kata/poker.clj [hunk 1]
```diff
@@ -166,18 +172,17 @@
...
-         high2 high2]
-    (cond
-      (= 0 (count high1)) :draw
-      (> (first high1) (first high2)) :win
-      (< (first high1) (first high2)) :lose
-      :else (recur (rest high1) (rest high2)))))
+  (let [cmp-result (->> (map - high1 high2)
+                        (find-first #(not (zero? %))))]
+    ; This is super funny way to use condp. I obviously misunderstand something
+    (condp apply (list cmp-result)
```
Comment (chrisza4) [inline src/poker_kata/poker.clj:181]: This is funny

File: src/poker_kata/poker.clj [hunk 2]
```diff
@@ -10,15 +10,21 @@
...
-    "a" 14
-    "j" 11
-    "q" 12
-    "k" 13
-    (parse-int val)))
+  (let [card-value-map {"a" 14
+                        "j" 11
+                        "q" 12
+                        "k" 13}]
+    (-> (get card-value-map (string/lower-case val))
```
Comment (vehas) [inline src/poker_kata/poker.clj]:
``` clojure
  (def card->val  {"j" 11, "q" 12, "k" 13, "a" 14})
  (defn card-value-number* [card-value]
      (card->val card-value (parse-int card-value)))
  (def card-value-number (comp card-value-number* string/lower-case))
```

Comment (vehas) [reply src/poker_kata/poker.clj]: function get, using keyword or hashmap as a function  all have the last argument that accepts default value when getting nil

Comment (chrisza4) [reply src/poker_kata/poker.clj]: Nice!!

**Pull request merged**